cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -219,6 +228,133 @@ def _add_conditional_storage_widgets(
                 widgets_list.append(graph_widget)
         return widgets_list
 
+    def custom_pcluster_metric_filter(self, metric_name, filter_pattern, custom_namespace):
+        """Adding custom metric filter from named tuple."""
+        metric_filter = logs.CfnMetricFilter(
+            scope=self.stack_scope,
+            id=metric_name + " Filter",
+            filter_pattern=filter_pattern,
+            log_group_name=self.cw_log_group_name,
+            metric_transformations=[
+                logs.CfnMetricFilter.MetricTransformationProperty(
+                    metric_namespace=custom_namespace,
```

Comment:
Instead of having a new namespace for every cluster created, how about using [dimensions](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Dimension)?
Dimensions are characteristics associated with a metric (like Metadata). In this  case, a ParallelCluster/Error metric can be  characterised by the cluster, the partition, etc.

Furthermore, if we use a unique namespace for each cluster, the user experience while browsing through the CloudWatch Metrics Page would be as follows for each cluster
```
ParallelCluster/Errors/ClusterA >>> Metrics with no dimensions >>> Metrics
...
```

We could instead publish the metrics to `ParallelCluster/Errors` namespace, then set the cluster as one of the Metric Dimensions e.g.
```
metric_transformations=[
    logs.CfnMetricFilter.MetricTransformationProperty(
    	...
    	dimensions=[logs.CfnMetricFilter.DimensionProperty(
    		key="cluster",
    		value=self.config.cluster_name
    	)],
    	...
```

Such that the user will see/use one namespace i.e. `ParallelCluster/Errors` which can then be classified by Dimension i.e. cluster. The experience would then be:

```
ParallelCluster/Errors >>> By Cluster >>> Metrics
```

---
It's similar to how EC2 has one namespace but you can view metrics "By InstanceId"

Reply:
Thank you for the suggestion. @HWong-sudo, let's test it out.

Reply:
encountered issues with adding dimensions. When copied from documentation, got message": "type object 'CfnMetricFilter' has no attribute 'DimensionProperty. MetricTransformation property also does not have dimension attribute within my local init. Plus in this cloudformation PR (https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/issues/64) they mention: 

"_The following announcement makes dimensions available for metric filters.

https://aws.amazon.com/about-aws/whats-new/2021/05/amazon-cloudwatch-logs-announces-dimension-support-for-metric-filters/

However, it appears to be missing **when specifying the resource through CloudFormation**. Getting the following error on stack events console.

Encountered unsupported property Dimensions
The Python SDK documentation correctly shows this as available, but missing in CloudFormation docs for CloudWatchLogs.
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/logs.html#CloudWatchLogs.Client.put_metric_filter

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-metricfilter.html_"

They mention that on 7/16/2022 ( https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/issues/64#issuecomment-1157913657) that the feature "_has issues related to deployment and traffic migration_" and the latest update on 7/25/2022 says that the "_feature is not available on eu-central-1_" There has been no further update as of 7/29/2022

Based on the above evidence, I conclude that the dimensions are not available to use as for CFN metricFilter's metric transformation property. Thus I will not be implementing this feature. It can be postponed to a future goal